x-pack/platform/plugins/shared/fleet/server/services/package_policy.ts

```diff
@@ -1377,15 +1472,106 @@ class PackagePolicyClientImpl implements PackagePolicyClient {
+            ', '
+          )}`
+      );
+
+      await Promise.all(
+        Object.entries(policyUpdatesBySpace).map(([updateSpaceId, spacePackagPolicyUpdates]) => {
+          logger.debug(
+            `Calling bulkUpdate for space [${updateSpaceId}] with [${spacePackagPolicyUpdates.length}] package policies`
+          );
+
+          // We use a scoped SO client here using the fleet app context services because each bulk update must be done
+          // with a properly scoped SO client. This should be safe, from an access standpoint since the code above
+          // used the SO client provided on input, which was used to `getById()` the package policies thus it has access.
+          return appContextService
+            .getInternalUserSOClientForSpaceId(updateSpaceId)
+            .bulkUpdate<PackagePolicySOAttributes>(spacePackagPolicyUpdates)
+            .catch(catchAndWrapError)
+            .then(({ saved_objects: updateResults }) => {
+              // For clarity and to help debug issues, we alter any error encountered an add info. about the space to it.
+              updateResults.forEach((updateResult) => {
+                if (updateResult.error) {
+                  updateResult.error.message += ` (for space [${updateSpaceId}])`;
+                }
+              });
+
+              packagePolicySoBulkUpdateResponse.push(...updateResults);
+            });
+        })
+      );
+    } else {
+      const updates = Object.values(policyUpdatesBySpace).flat();
+
+      logger.debug(() => `Calling SO bulkUpdate with [${updates.length}] updates`);
+
+      await soClient
+        .bulkUpdate<PackagePolicySOAttributes>(updates)
+        .catch(catchAndWrapError)
+        .then(({ saved_objects: updateResults }) => {
+          packagePolicySoBulkUpdateResponse.push(...updateResults);
+        });
+    }
+
+    logger.debug(
+      () =>
+        `Bulk update of the package policies SO documents done. Continuing to trigger version bump on associated agent policies`
     );
 
+    for (const updatedPolicyResult of packagePolicySoBulkUpdateResponse) {
+      if (updatedPolicyResult.error) {
+        const hasAlreadyFailed = failedPolicies.some(
+          (failedPolicy) => failedPolicy.packagePolicy.id === updatedPolicyResult.id
+        );
+        if (!hasAlreadyFailed) {
+          failedPolicies.push({
+            packagePolicy: packagePolicyUpdates.find((p) => p.id === updatedPolicyResult.id)!,
+            error: new FleetError(
+              `Bulk update of package policy [${updatedPolicyResult.id}] failed with ${updatedPolicyResult.error.message}`,
+              updatedPolicyResult.error
+            ),
+          });
+        }
+      } else {
+        updatedPoliciesSuccess.push(
+          mapPackagePolicySavedObjectToPackagePolicy(
+            updatedPolicyResult as SavedObject<PackagePolicySOAttributes>
+          )
+        );
+      }
+    }
+
```

Comment:
I was a bit concerned here when looking at the older code. We seem to continue on and trigger a agent policy version bump/deployment even if we have package policy SO update failures. Is that correct behavior?

I did not change this behavior with my changes, but it seems that perhaps we could be triggering policy deployment unnecessary.